client/src/components/Day.jsx

```diff
@@ -1,9 +1,70 @@
 import React from 'react';
+import style from '../scripts/style.css.js';
+import getMonthDayYear from '../scripts/getMonthDayYear.js';
 
-const Day = ({ day }) => {
-  { return day
-    ? <td>{day}</td>
-    : <td> </td>; }
+const Day = ({ day, vacancy, select, year, month, checkin, checkout, limit }) => {
+  if (vacancy !== undefined) {
+    let text = vacancy ? style.available : style.unavailable;
+    const today = new Date;
+    const nowMonth = today.getMonth();
+    const nowDay = today.getDate();
+    const nowYear = today.getFullYear();
+    const dayMonth = Number(month);
+    const dayDay = Number(day);
+    const dayYear = Number(year);
+    if (dayYear < nowYear 
```

Comment:
Do you think it is a good idea to convert these if checks to small functions with descriptive names? It would be easier to read and test. I had difficulty in following them

Reply:
Just trying to get functionality down, will refactor later!